connection_test.go

```diff
@@ -765,3 +766,35 @@ func TestConnectTimeout(t *testing.T) {
 		close(testComplete)
 	})
 }
+
+func TestTosPriority(t *testing.T) {
+	ctx, cancel := NewContext(time.Second)
+	defer cancel()
+
+	opts1 := testutils.NewOpts().SetServiceName("s1").SetTosPriority(tos.LOWDELAY).NoRelay()
```

Comment:
Couple of things:
 - you don't call "s1" so don't need to set the name
 - you don't actually need 2 listening servers here, only one's being called
 - we should add a comment for why relay is disabled here
 - no need to close, since servers created through the `TestServer` will automatically be closed
 - add context if it makes sense to assert/require.NoError
 - why does the WriteArgs need to happen in a goroutine?

I think this can actually be simplified to:
```go
// Relay is disabled because [....]
opts := testutils.NewOpts().SetServiceName("s1").SetTosPriority(..).NoRelay()
testutils.WithTestServer(t, opts, func(ts *testutils.TestServer) {
  ts.Register(raw.Wrap(...))

  client := ts.NewClient(tnil)
  outbound, err := client.BeginCall(ctx, ts.HostPort(), "s1", "echo", nil)
  require.NoError(t, err, "BeginCall failed")

  _, outboundNetConn := OutboundConnection(outbound)
  // verify conn TOS here.

  // Complete the call
  raw.WriteArgs(....)
})
```